client/client.go

```diff
@@ -1403,16 +1403,32 @@ func (c *Client) blockForRemoteAlloc(alloc *structs.Allocation) {
 		c.migratingAllocsLock.Unlock()
```

Comment:
Move the logger message at the top of the function such that we only show that if we are actually blocking